### PR TITLE
fix: default sourced by supplier to 1 in subcontracting order supplied items

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
default sourced by supplier to 1 in subcontracting order supplied items and update raw materials cost accordingly for finished good item.

### CODE REVIEW

File: erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.js [hunk 1]
```diff
@@ -154,11 +154,20 @@ frappe.ui.form.on('Subcontracting Order Supplied Item', {
...
 		let row = locals[cdt][cdn];
 		if (!row.sourced_by_supplier) {
-			frappe.db.get_value("Item", row.rm_item_code, "valuation_rate").then(r=> {
-				if (r.message.valuation_rate) {
-					row.rate = r.message.valuation_rate;
-					row.amount = row.rate * row.required_qty;
-					frm.events.calculate_rm_cost(frm);
+			const data = {
+				"item_code": row.rm_item_code
+			}
```
Comment (Alchez) [inline erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.js]: I think you can remove this one

File: erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.py [hunk 2]
```diff
@@ -234,3 +234,11 @@ def update_subcontracting_order_status(sco):
...
 
 	sco.update_status()
+
+
+@frappe.whitelist()
+def get_rm_valuation_rate(rm_item_code):
+
+	from erpnext.manufacturing.doctype.bom.bom import get_valuation_rate
+
+	return get_valuation_rate({"item_code": rm_item_code})
```
Comment (Alchez) [inline erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.py:244]: This should happen on the server-wide as well, when the document is saved

**Review by Alchez — CHANGES_REQUESTED** [review]

File: erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.py (hunk 2 above)
Comment (MyuddinKhatri) [reply erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.py:244]: Yes this happens when the document is saved via subcontracting_controller.

Comment (Alchez) [reply erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.py:244]: Oh awesome

**Review by Alchez — APPROVED** [review]

**Comment by Alchez** [comment]
@MyuddinKhatri, tests seem to be failing, can you check them out?

**Comment by MyuddinKhatri** [comment]
> @MyuddinKhatri, tests seem to be failing, can you check them out?

Sure I'll check.

**Pull request closed without merging**